test/blocks/merch-card/merch-card.test.js

```diff
@@ -153,6 +145,128 @@ describe('Merch Card', () => {
+      const buttons = footer.querySelectorAll('.con-button');
+
+      expect(merchCard).to.exist;
+      expect(heading).to.exist;
+      expect(headingOne).to.exist;
+      expect(body).to.exist;
+      expect(detail).to.exist;
+      expect(actionMenu).to.not.exist;
+      expect(merchCard.getAttribute('variant')).to.be.equal('catalog');
+      expect(merchCard.getAttribute('badge')).to.not.exist;
+      expect(merchCard.getAttribute('action-menu')).to.not.exist;
+      expect(merchCard.getAttribute('image')).to.exist;
+      expect(body.textContent).to.be.equal('Create gorgeous images, rich graphics, and incredible art. Save 10% for the first year. Ends Mar 20.See terms');
+      expect(detail.textContent).to.be.equal('Desktop + Mobile');
+      expect(buttons.length).to.be.equal(2);
+      expect(buttons[0].textContent).to.be.equal('Learn More');
+      expect(buttons[1].textContent).to.be.equal('Save now');
+    });
+
+    it('Supports Catalog card with badge without action-menu', async () => {
+      document.body.innerHTML = await readFile({ path: './mocks/catalog.html' });
+      await init(document.querySelector('.merch-card.catalog.empty-badge.action-menu-exist'));
+      const merchCard = document.querySelector('merch-card');
+      const heading = merchCard.querySelector('h3[slot="heading-xs"]');
+      const headingOne = merchCard.querySelector('h2[slot="heading-m"]');
+      const body = merchCard.querySelector('div[slot="body-xs"]');
+      const actionMenu = merchCard.querySelector('div[slot="action-menu-content"]');
+      const detail = merchCard.querySelector('h5[slot="detail-m"]');
+      const footer = merchCard.querySelector('div[slot="footer"]');
+      const buttons = footer.querySelectorAll('.con-button');
+
+      expect(merchCard).to.exist;
+      expect(heading).to.exist;
+      expect(headingOne).to.exist;
+      expect(body).to.exist;
+      expect(detail).to.exist;
+      expect(actionMenu).to.not.exist;
+      expect(merchCard.getAttribute('variant')).to.be.equal('catalog');
+      expect(merchCard.getAttribute('badge-text')).to.be.equal('LOREM IPSUM DOLOR');
+      expect(merchCard.getAttribute('badge-background-color')).to.be.equal('#EDCC2D');
+      expect(merchCard.getAttribute('badge-color')).to.be.equal('#000000');
+      expect(merchCard.getAttribute('action-menu')).to.not.exist;
+      expect(merchCard.getAttribute('image')).to.exist;
+      expect(body.textContent).to.be.equal('Create gorgeous images, rich graphics, and incredible art. Save 10% for the first year. Ends Mar 20.See terms');
+      expect(detail.textContent).to.be.equal('Desktop + Mobile');
+      expect(buttons.length).to.be.equal(2);
+      expect(buttons[0].textContent).to.be.equal('Learn More');
+      expect(buttons[1].textContent).to.be.equal('Save now');
+    });
+  });
+
   describe('UAR Card', () => {
     before(async () => {
       document.body.innerHTML = await readFile({ path: './mocks/uar-card.html' });
```

Comment:
A lot of these tests might be better served with snapshots. Not sure if we have implemented that or can in web test runner, but a shortcut would be have your input mock/output mock and deep equal.